### PR TITLE
Pitch Solo Extraction

### PR DESCRIPTION
**Pull request by ewega**
- Created sendPitch.py , gets only pitch from youtube URL
- Also edited import-clip code to run locally easliy (when testing)
- ALSO added progress prompts to main.py (if we ever even use it)

**TESTING**
We want to test locally, so for a random video (ideally kind of short just for testing) do the following. DJ Trump Video: "_https://www.youtube.com/watch?v=JZRXESV3R74_"

- Turn on localhost and everything
- Create "_**labelled**_" table locally (you might have to create the columns too idk)
- Add a "pitch" column to **_"labelled"_** table
- cd `import-clips`
- `npm start https://www.youtube.com/watch?v=JZRXESV3R74 log`
This should fill the local table with empty rows for all time intervals.

- cd `audio-analysis`
- `python sendPitch.py log`
- Enter the DJ Link above (put whatever for tag)
When finished running, the "pitch" column for that URL in **_"labelled"_** table should be filled out.

Might have dependencies and stuff.

NEXT STEPS: Run this code on our prod DB.

image

### CODE REVIEW

File: audio-analysis/lib/etl/dfHelper.py [hunk 1]
```diff
@@ -24,3 +24,17 @@ def customLeftJoin(dfLeft, dfRight, joinCondition):
...
+
+    sql = """
+            UPDATE labelled, temp_table 
+            SET labelled.pitch = temp_table.pitch 
+            WHERE labelled.url = temp_table.url 
+                AND labelled.start = temp_table.start_time_s;
+            """
+    engine.execute(sql)
+    sql = "DROP TABLE `temp_table`"
+    engine.execute(sql)
```
Comment (tjlam) [inline audio-analysis/lib/etl/dfHelper.py]: new line

File: audio-analysis/lib/etl/sqlConnection.py [hunk 2]
```diff
@@ -44,11 +44,20 @@ def sqlConnectionSetup(engine):
     
     meta.create_all() # Create all tables
 
+#Function checks if url exists in table already
+def urlExistsSpecific(url, engine, column_name):
+    # Check if URL already exists
+    result = engine.execute('SELECT * FROM `labelled` WHERE `url`="' + url + '" AND `pitch` IS NOT NULL')
```
Comment (tjlam) [inline audio-analysis/lib/etl/sqlConnection.py]: Should we pass in `column_name` into the sql query? or is it only used for the print statement?

File: audio-analysis/sendPitch.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+TESTING = False
+if len(sys.argv) == 2:
+    if sys.argv[1] == 'log':
+        TESTING = True
+
+
+engine = getEngine(TESTING)
+
+url = input("Provide YT Link: ")
+tag = input("Provide highlight tag (True/False): ")
```
Comment (tjlam) [inline audio-analysis/sendPitch.py]: Do we need this anymore?

File: audio-analysis/sendPitch.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# Pass `python main.py log` as optional arg to see print statements
+TESTING = False
+if len(sys.argv) == 2:
+    if sys.argv[1] == 'log':
+        TESTING = True
+
+
+engine = getEngine(TESTING)
+
+url = input("Provide YT Link: ")
```
Comment (tjlam) [inline audio-analysis/sendPitch.py]: Can we remove input prompts. I find its faster to add the yt url in the args. `sendPitch.py [yt url] log`

File: import-clips/run.js [hunk 5]
```diff
@@ -60,14 +74,29 @@ const OVERLAP_TIME = 2;
...
   fetchVideoInfo(videoId, (err, videoInfo) => {
     if (err) throw new Error(err);
     videoDuration = videoInfo.duration;
     console.log(videoDuration);
     const clipIntervals = getClipIntervals(videoDuration, CLIP_LENGTH, OVERLAP_TIME);
-    newRows = clipIntervals.map( interval => formatRow(youtubeUrl, interval))
-    sendToDB(newRows);
+    newRows = clipIntervals.map( interval => formatRow(youtubeUrl, interval));
+    if(TESTING){
+      con.connect(function(err) {
```
Comment (tjlam) [inline import-clips/run.js]: I should have done it this way to start but I was lazy and used the flask app end point. Let's make this into a method, and we can use it to update local or prod.

File: audio-analysis/lib/etl/sqlConnection.py (hunk 2 above)
Comment (tjlam) [reply audio-analysis/lib/etl/sqlConnection.py]: Why do we check if pitch is not null? If it is null we still want to update it right?

**Review by tjlam — COMMENTED** [review]
Looks good, mostly some nits and questions

**Comment by ewega** [comment]
Made the fixes.

File: audio-analysis/lib/utils/videoHelper.py [hunk 6]
```diff
@@ -67,7 +66,8 @@ def getFilename(self):
 
     def getFilenameWav(self):
         video_title = self.info_dict.get('title', None)
-        filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
+        #filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
+        filename_wav = 'audio-files/CS -GO - Fnatic vs. mousesports [Inferno] Map 1 - Grand Final - ESL Pro League Season 10 Finals-0AlFcix7aaY.wav'
```
Comment (tjlam) [inline audio-analysis/lib/utils/videoHelper.py]: is this a test file?

File: audio-analysis/lib/etl/sqlConnection.py [hunk 7]
```diff
@@ -44,11 +44,20 @@ def sqlConnectionSetup(engine):
     
     meta.create_all() # Create all tables
 
+#Function checks if url exists in table already
+def urlExistsSpecific(url, engine, column_name):
```
Comment (mattlianje) [inline audio-analysis/lib/etl/sqlConnection.py]: helpful!

File: audio-analysis/lib/etl/sqlConnection.py [hunk 8]
```diff
@@ -44,11 +44,20 @@ def sqlConnectionSetup(engine):
...
+    # Check if URL already exists
+    result = engine.execute('SELECT * FROM `labelled` WHERE `url`="' + url + '" AND `'+ column_name +'` IS NOT NULL')
```
Comment (tjlam) [inline audio-analysis/lib/etl/sqlConnection.py:50]: I still don't understand the `column_name IS NOT NULL` part of the query. Why are we filtering by not null? If we import a video by default the pitch would be null so this method would return false

**Review by tjlam — DISMISSED** [review]

File: audio-analysis/lib/utils/videoHelper.py [hunk 9]
```diff
@@ -67,7 +66,8 @@ def getFilename(self):
 
     def getFilenameWav(self):
         video_title = self.info_dict.get('title', None)
-        filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
+        #filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
```
Comment (mattlianje) [inline audio-analysis/lib/utils/videoHelper.py]: let s avoid wildcards and have this for all audio files

**Review by mattlianje — CHANGES_REQUESTED** [review]
LGTM) one lil one)

**Comment by tjlam** [comment]
You can add your secrets file to the import-clips folder. Just make sure you update `.gitignore` with `import-clips/secretsJS.js`

File: audio-analysis/lib/etl/sqlConnection.py (hunk 8 above)
Comment (mattlianje) [reply audio-analysis/lib/etl/sqlConnection.py:50]: Aaaaaa makes sense. @tjlam @Eldrick19 expained it to me. Basically it just checks if `column_name` has been filled in for a given `url` but the name of the method is a bit exotic.

Comment (tjlam) [reply audio-analysis/lib/etl/sqlConnection.py:50]: Why do we need to check if that column is filled though? If this returns false we don't want to update right?

Comment (ewega) [reply audio-analysis/lib/etl/sqlConnection.py:50]: @tjlam You're right. I think it helps looking at this part of the code:
```
url_exists = urlExistsSpecific(url, engine, column_name)
if url_exists == True:
    print("This video is already in the database. Will overwrite ", column_name, " data.")
```

It should say "This video's pitch is already in the database" or whatever - mb on wording there. But essentially its just for logging.

Comment (tjlam) [reply audio-analysis/lib/etl/sqlConnection.py:50]: Let's say that url is not in the database at all, would `updateDBWithDataFromDF` still work?

Comment (tjlam) [reply audio-analysis/lib/etl/sqlConnection.py:50]: I assume it wouldn't??? because it would have to insert new rows

Comment (ewega) [reply audio-analysis/lib/etl/sqlConnection.py:50]: That is correct, it assumes the URL is already in the DB

**Review by mattlianje — APPROVED** [review]
`bang`

**Pull request merged**